go/ct/st/transaction_context_test.go

```diff
@@ -43,16 +46,68 @@ func TestTransactionContext_String(t *testing.T) {
 			t.OriginAddress[19] = 0xfe
 			return t.OriginAddress
 		}},
+		"Blob Hashes": {func(t *TransactionContext) any {
+			t.BlobHashes = []vm.Hash{{1, 2, 3, 4}}
+			return t.BlobHashes
+		}},
 	}
 
-	c := TransactionContext{}
 	for name, test := range tests {
 		t.Run(name, func(t *testing.T) {
+			c := TransactionContext{}
 			v := test.change(&c)
-			str := c.String()
-			if !strings.Contains(str, fmt.Sprintf("%v: %v", name, v)) {
+			got := c.String()
+			want := fmt.Sprintf("%v: %v", name, v)
+			if !strings.Contains(got, want) {
 				t.Errorf("Did not find %v string", name)
 			}
 		})
 	}
 }
+
+func TestTransactionContext_Clone(t *testing.T) {
+
+	t1 := TransactionContext{
+		OriginAddress: vm.Address{1, 2, 3, 4},
+		BlobHashes:    []vm.Hash{{1, 2, 3, 4}},
+	}
+
+	t2 := t1.Clone()
+
+	if !t1.Equal(t2) {
+		t.Errorf("Cloned transaction context is not equal to original")
+	}
+}
+
+func TestTransactionContext_Equal(t *testing.T) {
+
+	tests := map[string]struct {
+		change func(*TransactionContext, *TransactionContext)
```

Comment:
Why the change function instead of two fields `a` and `b`?

Reply:
I thought change function conveyed the right meaning in this case, to modify depending on the test case.

Reply:
A test for an `Equal` function should be about pairs of values as they are at the time of the comparison, not about the historic evolution how they become to be what hey are today -- what is implied with the name `change` and the way changes are used in other tests.

Also, using a pairs of values would make the code much shorter and simpler:
```
	tests := map[string]struct {
		a, b *TransactionContext
	}{
		"zero-values": {
			a: &TransactionContext{},
			b: &TransactionContext{},
		},
		"match": {
			a: &TransactionContext{BlobHashes: []vm.Hash{{1}}},
			b: &TransactionContext{BlobHashes: []vm.Hash{{1}}},
		},
		"mismatch": {
			a: &TransactionContext{BlobHashes: []vm.Hash{{1}}},
			b: &TransactionContext{BlobHashes: []vm.Hash{{4}}},
		},
		"mismatch-different-blob-hash-length": {
			a: &TransactionContext{BlobHashes: []vm.Hash{{1}, {2}}},
			b: &TransactionContext{BlobHashes: []vm.Hash{{4}}},
		},
	}

	for name, test := range tests {
		t.Run(name, func(t *testing.T) {
			a, b := test.a, test.b
			if got, want := a.Eq(b), reflect.DeepEqual(a, b); got != want {
				t.Errorf("failed to compare contexts %v and %v, wanted %t, got %t", a, b, want, got)
			}
		})
	}
```